multigeiger/thp_sensor.cpp

```diff
@@ -4,29 +4,61 @@
 
 #include <Adafruit_Sensor.h>
 #include <Adafruit_BME280.h>
+#include <Adafruit_BME680.h>
 
 #include "log.h"
 #include "thp_sensor.h"
 
-static bool have_thp = false;
+static bool have_thp;
+static int type_thp = 0;
 
-Adafruit_BME280 bme;
+Adafruit_BME280 bme280;
+Adafruit_BME680 bme680;
 
 bool setup_thp_sensor(void) {
-  have_thp = (bool) bme.begin(BME280_ADDRESS);
+  // BME280
+  have_thp = (bool) bme280.begin(BME280_ADDRESS);
   if (!have_thp)
-    have_thp = (bool) bme.begin(BME280_ADDRESS_ALTERNATE);
-  if (have_thp)
-    log(INFO, "BME_Status: %d  ID:%0X", (int) have_thp, bme.sensorID());
-  else
-    log(INFO, "BME_Status: %d  ID:None", (int) have_thp);
+    have_thp = (bool) bme280.begin(BME280_ADDRESS_ALTERNATE);
+  if (have_thp) {
+    type_thp = 2;
+    log(INFO, "BME_Status: %d  ID: BME280", (int) have_thp);
+    return have_thp;
+  }
+  // BME680
+  have_thp = (bool) bme680.begin(BME680_I2C_ADDR_PRIMARY);
+  if (!have_thp)
+    have_thp = (bool) bme680.begin(BME680_I2C_ADDR_SECONDARY);
+  if (have_thp) {
+    // Set up oversampling and filter initialization
+    bme680.setTemperatureOversampling(BME680_OS_8X);
+    bme680.setHumidityOversampling(BME680_OS_2X);
+    bme680.setPressureOversampling(BME680_OS_4X);
+    bme680.setIIRFilterSize(BME680_FILTER_SIZE_3);
+    bme680.setGasHeater(300, 150); // 300*C for 150 ms
```

Comment:
is this needed even if you do not use the gas sensor yet?

Reply:
I prefer to have the sensor fully initialized. For the portable version, an air quality index might be interesting, too...